test/e2e/pages/tabbed-page.ts

```diff
@@ -0,0 +1,19 @@
+import { Locator } from '@playwright/test';
+import { BasePage, GotoOptions } from './base-page';
+
+export abstract class TabbedBasePage<T extends TabbedBasePage<T>> extends BasePage<T> {
```

Comment:
I just learned that all of these generics were unnecessary, because in TypeScript you can use `this` as a type. e.g.:
```
async goto(options?: GotoOptions): Promise<this> {
   return this;
}
```
I'll improve that in another PR. I'm still cleaning up the user-management code.